store/tikv/region_cache_test.go

```diff
@@ -260,21 +303,22 @@ func (s *testRegionCacheSuite) TestReconnect(c *C) {
 }
 
 func (s *testRegionCacheSuite) TestRequestFail(c *C) {
+	ts := time.Now().Unix()
 	region := s.getRegion(c, []byte("a"))
-
 	ctx, _ := s.cache.GetRPCContext(s.bo, region.VerID())
-	s.cache.DropStoreOnSendRequestFail(ctx, errors.New("test error"))
-	c.Assert(s.cache.mu.regions, HasLen, 0)
+	s.cache.OnSendRequestFail(ctx, errors.New("test error"))
 	region = s.getRegion(c, []byte("a"))
 	c.Assert(s.cache.mu.regions, HasLen, 1)
 	ctx, _ = s.cache.GetRPCContext(s.bo, region.VerID())
-	s.cache.DropStoreOnSendRequestFail(ctx, errors.New("test error"))
-	c.Assert(len(s.cache.mu.regions), Equals, 0)
+	s.cache.OnSendRequestFail(ctx, errors.New("test error"))
+	c.Assert(workableRegions(s.cache, s.cache.mu.regions, ts), Equals, 0)
+	time.Sleep(2 * time.Second)
```

Comment:
plz do not sleep

Reply:
have refined test case and maybe we still need sleep for test blackout store logic